### ISSUE TITLE
Saving generates: Attempt to de-reference a null object

### ISSUE THREAD

**Issue by softwareangel** [Kathy-Lueckeman/EASY-Open-Source-Solution#4]
Greetings,

After reviewing the Installation, User and Technical Guides, I am not able to resolve this error within my Developer Org.  I have a simple interface with Requirement Items of an Address and Document Upload, which allows input.  
![image](https://user-images.githubusercontent.com/45464122/49285705-2c303b00-f466-11e8-8ba7-2f964c9a1911.png)

![image](https://user-images.githubusercontent.com/45464122/49286281-a1e8d680-f467-11e8-8368-9d739ae37923.png)

I have tried these Requirement Items active and inactive, but the results generate an error after clicking the "Save" button, then "Processing" displays, then "Error: Attempt to de-reference a null object".
![image](https://user-images.githubusercontent.com/45464122/49285740-4f5aea80-f466-11e8-9b0b-186db2afacab.png)

I attempted to debug, but the debug logs report success without a clue of what apex code is experiencing the de-referencing (i.e. ApplicationRegistration.apex)

These are my Custom Settings and URL for portal.  Any help would be appreciated.

![image](https://user-images.githubusercontent.com/45464122/49286011-21c27100-f467-11e8-93ea-4f268c143ab2.png)

Portal URL = https://thinkkentucky-developer-edition.na50.force.com/directory/ApplicationRegistration

**Comment by kyleschmid** [comment]
@softwareangel The debug logs aren't reporting an error because it's being caught in a try catch block, so the entire operation is successful. You should be able to scroll up and see what the error message is, however. My guess with this one is that there's a problem creating the user. This normally comes from three things:

1. The Interaction Mappings aren't set-up to create a Contact
2. The Site Contact doesn't have a role (portal account must have a role error)
3. The Application Community permission set has the "Create" permission on Account and you have regular Customer Community Login licenses and not Customer Community Login Plus

It could be something else, however. You'll want to take a look at the debug logs to confirm.

**Comment by softwareangel** [comment]
Thanks for your reply.

   1. The Interaction Mappings are set-up to create a Contact according to
   the Installation instructions.
   2. The Site Contact doesn't have a role (portal account must have a role
   error)... This configuration was not part of the Installation instructions.
   I will revisit.
   3. The Application Community permission set has the "Create" permission
   on Account and you have regular Customer Community Login licenses and not
   Customer Community Login Plus.... I will switch to the Customer Community
   Login Plus (I have the "guest" user profile code, which was based on the
   instructions.)

I will let you know what happens when I revisit these suggestions.

Thanks.

Raul R. Edwards - smartphone
ENOVATIONAL CORP

**Comment by softwareangel** [comment]
Greetings,

Success!!!!  I revisited the following suggestions:

1. The Interaction Mappings were set-up to create a Contact according to the installation instructions.
2. The Site Contact (in Settings > Users > Roles) had a role set already from when the Developer Org was created.
3. In the Settings > Custom Settings > Application Settings, I switched from the "guest" profile (obtained from All Communities > Workspace > Administration > Go to Force.com > Public Access Settings) to the Custom Community Plus Login User profile.
4. Added the Custom Community Plus Login User to the Profile list: All Communities > Workspace > Administration > Members > Selected Profile
5. Selected Custom Community Plus Login User in the All Communities > Workspace > Administration > Login & Registration > Registration Configuration Page section.

The key to the solution was changing to the "Custom Community Plus Login User" profile/license.

Thanks for your assistance.
